packages/hot-module-replacement/client.js

```diff
@@ -487,35 +491,24 @@ Meteor.startup(function () {
     if (nonRefreshableVersion !== doc.versionNonRefreshable) {
       nonRefreshableVersion = doc.versionNonRefreshable;
       console.log('HMR: Some changes can not be applied with HMR. Using hot code push.')
-      mustReload = true;
-      if (pendingReload) {
-        pendingReload();
-      }
+      forceReload();
     } else if (doc.versionReplaceable !== replaceableVersion) {
       replaceableVersion = doc.versionReplaceable;
-      if (!mustReload) {
-        if (pendingReload) {
-          requestChanges();
-        }
-      } else {
-        mustReload = true;
-        if (pendingReload) {
-          pendingReload();
-        }
+
+      if (useHotCodePush) {
+        return forceReload();
       }
+
+      requestChanges();
     }
   });
 
-  // We disable hot code push for js until there were
-  // changes that can not be applied through HMR.
-  Package['reload'].Reload._onMigrate(function (tryReload) {
-    if (mustReload) {
+  Package['reload'].Reload._onMigrate(function () {
```

Comment:
Is it safe to assume that every app has this package?

Reply:
Yes. It is a dependency of `hot-code-push` which `hot-module-replacement` depends on. The code here is getting it from the Packages object because the Reload package loads after hot-module-replacement.